apis/v1alpha1/gatewayclass_types.go

```diff
@@ -62,24 +62,18 @@ type GatewayClassSpec struct {
 	// +required
 	Controller string `json:"controller"`
 
-	// AllowedGatewayNamespaceSelector is a selector of namespaces that Gateways
-	// can use this GatewayClass from. This is a standard Kubernetes
-	// LabelSelector. Controllers must not support Gateways in namespaces
-	// outside this selector.
-	//
-	// An empty selector (default) indicates that Gateways can use this
-	// GatewayClass from any namespace.
-	//
-	// When a Gateway attempts to use this class from a namespace that is not
-	// allowed by this selector, the controller implementing the GatewayClass
-	// may add a new "ForbiddenNamespaceForClass" condition to the Gateway
-	// status. Adding this condition is considered optional since not all
-	// controllers will have access to all namespaces.
+	// AllowedGatewayNamespaces is a selector of namespaces that Gateways of
+	// this class can be created in. Implementations must not support Gateways
+	// when they are created in namespaces not specified by this field. If
+	// Gateways have already been provisioned when this field changes, they will
+	// continue to be considered valid and supported. For the purposes of this
+	// field, a Gateway must only be considered provisioned if it appears in the
+	// ProvisionedGateways list in GatewayClassStatus.
 	//
 	// Support: Core
 	//
 	// +optional
-	AllowedGatewayNamespaceSelector metav1.LabelSelector `json:"allowedGatewayNamespaceSelector"`
+	AllowedGatewayNamespaces metav1.LabelSelector `json:"allowedGatewayNamespaces"`
```

Comment:
Add `omitempty` to json tag if this is optional.

Reply:
@robscott since `AllowedGatewayNamespaces` is optional, is the default behavior to allow all/no namespaces?

Reply:
I think I should update this to be an explicit default of `{}`. I'd thought that's what optional without omitempty would do, but even if it does, that's not as clear as actually setting a default value here. The intent is for the default to allow a GatewayClass to be used everywhere - I imagine that will be the most common use case.

Reply:
Actually, I thought about it more and opted to make it empty and just define the unspecified state like you recommended.